connector/src/main/java/org/geysermc/connector/network/translators/inventory/translators/PlayerInventoryTranslator.java

```diff
@@ -272,11 +290,23 @@ public SlotType getSlotType(int javaSlot) {
                     if (!(checkNetId(session, inventory, swapAction.getSource()) && checkNetId(session, inventory, swapAction.getDestination()))) {
                         return rejectRequest(request);
                     }
-                    if (isCraftingGrid(swapAction.getSource()) || isCraftingGrid(swapAction.getDestination())) {
-                        return rejectRequest(request, false);
+                    if (isCraftingGrid(swapAction.getSource())) {
+                        return rejectRequest(request);
                     }
 
-                    if (isCursor(swapAction.getDestination())) {
+                    if (isCraftingGrid(swapAction.getDestination())) {
+                        int sourceSlot = bedrockSlotToJava(swapAction.getSource());
+                        int destSlot = bedrockSlotToJava(swapAction.getDestination());
+
+                        inventory.setItem(sourceSlot, GeyserItemStack.EMPTY, session);
+                        inventory.setItem(destSlot, GeyserItemStack.from(ItemTranslator.translateToJava(UNUSUABLE_CRAFTING_SPACE_BLOCK)), session);
```

Comment:
We should never set the persistent inventory to have the unusable crafting space block.